### PR TITLE
Update example copilot to use SSEs + refactor

### PR DESCRIPTION
**Pull request by mnicstruwig**
Updates the base `example-copilot` to use `magentic` and SSEs.

I also add a very basic test suite (with the intention, later on, after running all of these tests as part of a Github Action). 

If everyone is happy with this pattern, I will apply it to the other example copilots.

### CODE REVIEW

File: example-copilot/example_copilot/models.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from typing import Any
+from uuid import UUID
 from pydantic import BaseModel, Field, validator
```
Comment (montezdesousa) [inline example-copilot/example_copilot/models.py:3]: "Pydantic V1 style `@validator` validators are deprecated. You should migrate to Pydantic V2 style `@field_validator` validators, see the migration guide for more details"

Should be just swapping `validator` to `field_validator`

**Review by montezdesousa — CHANGES_REQUESTED** [review]

File: example-copilot/example_copilot/models.py (hunk 1 above)
Comment (mnicstruwig) [reply example-copilot/example_copilot/models.py:3]: That's a great point, and a good catch. Thank you! Will update.

File: example-copilot/README.md
Comment (montezdesousa) [inline example-copilot/README.md:1]: If we are suggesting `poetry install --no-root`, `poetry` creates its own environment the start command needs to be `poetry run example_copilot.main:app --port 7777 --reload`.

**Review by montezdesousa — CHANGES_REQUESTED** [review]

**Comment by montezdesousa** [comment]
Tested this at https://pro.openbb.co, no issues besides that README detail.

**Review by IgorWounds — APPROVED** [review]
Aside from what @montezdesousa mentioned, this LGTM.

**Review by montezdesousa — APPROVED** [review]

**Pull request merged**